homeassistant/components/goalzero/__init__.py

```diff
@@ -40,14 +39,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     coordinator = GoalZeroDataUpdateCoordinator(hass, api)
     await coordinator.async_config_entry_first_refresh()
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+    entry.runtime_data = coordinator
```

Comment:
Part of the benefit of using runtime_data is the ability to type it.

https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry

Reply:
Addressed in 2014241398e62d622a41edac6de8ba747238ad64